File: AppMovilGestionCreditos1/lib/models/usuario_dto.dart
import 'cliente_dto.dart';

class UsuarioDTO {
  int id;
  String? nombreApellidos;
  String? correo;
  int? rolId ;
  String? rolDescripcion;
  String? clave;
  int? esActivo;
  ClienteDTO? cliente; // Relación anidada para registro completo

  UsuarioDTO({
    this.id = 0,
    this.nombreApellidos,
    this.correo,
    this.rolId,
    this.rolDescripcion,
    this.clave,
    this.esActivo,
    this.cliente,
  });

   // ---- FROM JSON ----
  factory UsuarioDTO.fromJson(Map<String, dynamic> json) {
    return UsuarioDTO(
      id: json["id"] ?? 0,
      nombreApellidos: json["nombreApellidos"],
      correo: json["correo"],
      //rolId: json["rolId"],
      rolDescripcion: json["rolDescripcion"],
      clave: json["clave"],
      esActivo: json["esActivo"],
      cliente: json["cliente"] != null
          ? ClienteDTO.fromJson(json["cliente"])
          : null,
    );
  }

  // ---- TO JSON ----
  Map<String, dynamic> toJson() => {
        'Id': id,
        'NombreApellidos': nombreApellidos,
        'Correo': correo,
        'RolId': rolId,
        'RolDescripcion': rolDescripcion,
        'Clave': clave,
        'EsActivo': esActivo,
        'Cliente': cliente?.toJson(),
      };
}

File: ApiCredito2-main/GestionIntApi/DTO/UsuarioDTO.cs
using Microsoft.AspNetCore.Mvc;
using System.ComponentModel.DataAnnotations;

namespace GestionIntApi.DTO
{
    public class UsuarioDTO 
    {
        public int Id { get; set; }
        [Required(ErrorMessage = "El nombre y apellido es obligatorio")]
        [MinLength(3, ErrorMessage = "El nombre debe tener al menos 3 caracteres")]
        public string? NombreApellidos { get; set; }

        [Required(ErrorMessage = "El correo es obligatorio")]
        [EmailAddress(ErrorMessage = "El correo no es válido")]
        public string? Correo { get; set; }
        public int? RolId { get; set; }
        public String? RolDescripcion { get; set; }

        [Required(ErrorMessage = "La clave es obligatoria")]
        [MinLength(6, ErrorMessage = "La clave debe tener mínimo 6 caracteres")]
        public string? Clave { get; set; }
        public int? EsActivo { get; set; }


        public ClienteDTO Cliente { get; set; }
    }
}
